File: windows/src/GoalMaker.Core/Sync/SyncStatus.cs
namespace GoalMaker.Core.Sync;

/// <summary>The device's sync state, the last success, and how many changes are still local.</summary>
public sealed record SyncStatus(SyncState State, DateTimeOffset? LastSyncedAt, int PendingChanges, string? Problem)
{
    public static SyncStatus Initial { get; } = new(SyncState.Idle, null, 0, null);
}

File: android/app/src/main/kotlin/com/goalmaker/app/application/sync/SyncStatus.kt
package com.goalmaker.app.application.sync

import java.time.Instant

/** The device's sync state, the last success, and how many changes are still local. */
data class SyncStatus(
    val state: SyncState,
    val lastSyncedAt: Instant?,
    val pendingChanges: Int,
    val problem: String?,
) {
    companion object {
        val INITIAL = SyncStatus(SyncState.IDLE, lastSyncedAt = null, pendingChanges = 0, problem = null)
    }
}
